crates/rust-analyzer/src/handlers/request.rs

```diff
@@ -548,6 +549,10 @@ pub(crate) fn handle_will_rename_files(
 ) -> anyhow::Result<Option<lsp_types::WorkspaceEdit>> {
     let _p = profile::span("handle_will_rename_files");
 
+    if let Err(err) = resource_ops_supported(&snap.config, ResourceOperationKind::Rename) {
+        return Err(err);
+    }
+
```

Comment:
```suggestion
```
This request istriggered by the client when it is about to rename a file, so we only emit text edits here, not file edits. No need to check this capabality here